### ISSUE TITLE
TestResult should include more information about errors

### ISSUE THREAD

**Issue by lzybkr** [pester/Pester#457]
When there is an error, the TestResult object has a FailureMessage, but is missing some possibly useful information such as the FullyQualifiedErrorId or the actual exception type.

This additional information is useful when testing localized builds - the failure message may be localized, but the FullyQualifiedErrorId and exception type are not.

For example, if my test was:

```
Describe "Test SubString" {
    "abc".Sbustring(0,1) | Should Be a
}
```

And running the test:

```
#236 PS> $x = Invoke-Pester -PassThru .\b.ps1
Describing Test SubString
 [-] Error occurred in Describe block 26ms
   Method invocation failed because [System.String] does not contain a method named 'Sbustring'.
   At D:\xtmp\b.ps1:3 char:5
Tests completed in 26ms
Passed: 0 Failed: 1 Skipped: 0 Pending: 0
#237 PS> $x.TestResult


Describe               : Test SubString
Context                :
Name                   : Error occurred in Describe block
Result                 : Failed
Passed                 : False
Time                   : 00:00:00.0261851
FailureMessage         : Method invocation failed because [System.String] does not contain a method named 'Sbustring'.
StackTrace             : At D:\xtmp\b.ps1:3 char:5
ParameterizedSuiteName :
Parameters             :
```

If I want to automation around this failure that works with localized builds, FailureMessage won't work well, but if there was a property FullyQualifiedErrorId (which you can get from the ErrorRecord object), I could.

**Comment by dlwyatt** [comment]
Easy enough.  Since this is for the -PassThru object (all PowerShell, in memory), how about we just stick the whole ErrorRecord object onto the TestResult object?  This would only apply in cases where exceptions came from something other than the Should command (with the exception of `Should Throw`), but still includes a lot of extra information in its original form.

**Comment by lzybkr** [comment]
Sure, sounds good to me.

**Comment by dlwyatt** [comment]
Looks like we already made this change.  What version of Pester were you running in your test?

**Comment by dlwyatt** [comment]
Just checked the history, this came in #426 from @alx9r , first released in v3.3.12.  (Which had some bugs; use 3.3.14 instead.)

**Comment by lzybkr** [comment]
Yeah, I'm stuck on 3.3.5 (what's shipping with Windows 10).  It's good to know it's fixed in newer versions.
